evalml/pipelines/components/component_base.py

```diff
@@ -151,3 +151,18 @@ def __eq__(self, other):
             if getattr(self, attribute) != getattr(other, attribute):
                 return False
         return True
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        rpr = f"{(type(self).__name__)}("
+        for parameter, value in self.parameters.items():
+            if type(value) == str:
+                rpr = rpr + f"{parameter}='{value}',"
+            elif value == float('inf') or value == float('-inf'):
+                rpr = rpr + f"{parameter}=float('{value}'),"
+            else:
+                rpr = rpr + f"{parameter}={value},"
+        rpr = rpr + ')'
+        return rpr
```

Comment:
@eccabay good call looking into `float('inf')`!! Another one is `float('nan')`. You've hit on [what appears to be a funny bug and/or design flaw with python's `float` type](https://bugs.python.org/issue22951): `repr(float('inf'))` comes back as `'inf'`, which... is just messed up! 🤔   Same for `float('nan')`. I think they did this so that `float(repr(float('inf')))` is equivalent to `float('inf')`, and same for nan.

The same is true for `np.inf` and `np.nan`, by the way:
```python
In [6]: parameters = {'int': 42, 'string': 'string', 'float': 3.14159,
                      'inf': float('inf'), 'np inf': np.inf, 'np nan': np.nan}

In [7]: repr(parameters)
Out[7]: "{'int': 42, 'string': 'string', 'float': 3.14159, 'inf': inf, 'np inf': inf, 'np nan': nan}"
```

---
So, what do we do? It would be really nice if we could have what comes out of `repr(component)` be something we could paste into a terminal in order to get an identical instance.

I like the approach you're taking. I played with it for a bit and here's what I got:
```python
# define this in gen_utils so we can use it for the pipeline repr too
def safe_repr(value):
    if isinstance(value, float):
        if pd.isna(value):
            return 'np.nan'
        if np.isinf(value):
            return f'float({repr(value)})'
    return repr(value)

# then the component repr:
def __repr__(self):
    parameters_repr = ', '.join([f'{key}={safe_repr(value)}' for key, value in self.parameters.items()])
    return f'{self.name}({parameters_repr})'
```

Here's a test I did of the `safe_repr`, which shows that it outputs the string we need in order to avoid some ugly bugs:
```
In [1]: [safe_repr(el) for el in ['string', float('nan'), float('-inf'), float('inf'), np.nan, -np.inf, np.inf]]
Out[1]:
["'string'", 'np.nan', 'float(-inf)', 'float(inf)', 'np.nan', 'float(-inf)', 'float(inf)']
```

What do you think of that? Weird stuff!